File: c++/S01-elemental-actions/E01-legal-data.cpp
#include <iostream>

int main(int argc, char *argv[]) {
	std::cout << "\n\e[0;35m[========= LEGAL DATA =========]\e[0m\n\n";

	std::cout << "Enter the following data:\n";

	std::string name;
	std::cout << " - Full Name: ";
	getline(std::cin, name);

	int age;
	std::cout << " - Age: ";
	std::cin >> age;

	char gender;
	std::cout << " - Gender (M/F): ";
	std::cin >> gender;

	float height;
	std::cout << " - Height (m): ";
	std::cin >> height;

	double weight;
	std::cout << " - Weight (kg): ";
	std::cin >> weight;

	bool isMarried;
	std::cout << " - Are you married? (1/0): ";
	std::cin >> isMarried;

	std::cout << "\n\e[0;35mCOLLECTED DATA\e[0m\n\n";

	std::cout << "Your full name:     \e[0;32m" << name << "\e[0m\n";
	std::cout << "Your age:           \e[0;32m" << age << "\e[0m\n";
	std::cout << "Your gender:        \e[0;32m" << gender << "\e[0m\n";
	std::cout << "Your height:        \e[0;32m" << height << "m\e[0m\n";
	std::cout << "Your weight:        \e[0;32m" << weight << "kg\e[0m\n";
	std::cout << "Are you married?:   \e[0;32m" << (isMarried ? "Yes" : "No") << "\e[0m\n";

	return 0;
}

File: java/E01_legal_data.java
import java.util.Scanner;

public class E01_legal_data {
    public static void main(String[] args) {
        System.out.println("\n\033[0;35m[========= LEGAL DATA =========]\033[0m\n");

		Scanner scanner = new Scanner(System.in);

		System.out.print("\n\033[0;34mEnter the following data: \033[0m\n");

		System.out.print(" - Enter your full name: ");
		String fullName = scanner.nextLine();

		System.out.print(" - Enter your age: ");
		int age = scanner.nextInt();

		System.out.print(" - Enter your gender [m/f]: ");
		char gender = scanner.next().charAt(0);

		System.out.print(" - Enter your height: ");
		double height = scanner.nextDouble();

		System.out.print(" - Enter your weight: ");
		double weight = scanner.nextDouble();

		System.out.print(" - Are you married? (true/false): ");
		boolean isMarried = scanner.nextBoolean();

        scanner.close();

		System.out.print("\n\033[0;34mCollected data: \033[0m\n");
		System.out.println("\033[0m - Full Name:    \033[0;32m" + fullName);
		System.out.println("\033[0m - Age:          \033[0;32m" + age);
		System.out.println("\033[0m - Gender:       \033[0;32m" + gender);
		System.out.println("\033[0m - Height:       \033[0;32m" + height);
		System.out.println("\033[0m - Weight:       \033[0;32m" + weight);
		System.out.println("\033[0m - is Married?:  \033[0;32m" + isMarried);
		System.out.println();
    }
}
